images/BUILD

```diff
@@ -7,128 +7,95 @@ load(
-        "//:etcd-v3.3.10-linux-amd64_etcd",
-    ],
-)
-
 # Layer for etcd 3.3.13, updated recommendation for k8s 1.14 and later
-container_layer(
-    name = "etcd-3-3-13-layer",
-    directory = "/opt/etcd-v3.3.13-linux-amd64/",
-    files = [
-        "//:etcd-v3.3.13-linux-amd64_etcdctl",
-        "//:etcd-v3.3.13-linux-amd64_etcd",
-    ],
-)
-
 # Layer for etcd 3.3.17, updated recommendation for k8s 1.16.3 and later
-container_layer(
-    name = "etcd-3-3-17-layer",
-    directory = "/opt/etcd-v3.3.17-linux-amd64/",
-    files = [
-        "//:etcd-v3.3.17-linux-amd64_etcdctl",
-        "//:etcd-v3.3.17-linux-amd64_etcd",
-    ],
-)
-
 # Layer for etcd 3.4.3, updated recommendation for k8s 1.17 and later
-container_layer(
-    name = "etcd-3-4-3-layer",
-    directory = "/opt/etcd-v3.4.3-linux-amd64/",
-    files = [
-        "//:etcd-v3.4.3-linux-amd64_etcdctl",
-        "//:etcd-v3.4.3-linux-amd64_etcd",
-    ],
-)
-
 # Layer for etcd 3.4.13, updated recommendation for k8s 1.19 and later
-container_layer(
-    name = "etcd-3-4-13-layer",
-    directory = "/opt/etcd-v3.4.13-linux-amd64/",
-    files = [
-        "//:etcd-v3.4.13-linux-amd64_etcdctl",
-        "//:etcd-v3.4.13-linux-amd64_etcd",
-    ],
-)
-
-load("@io_bazel_rules_docker//docker/package_managers:download_pkgs.bzl", "download_pkgs")
-load("@io_bazel_rules_docker//docker/package_managers:install_pkgs.bzl", "install_pkgs")
-
+[
+    container_layer(
+        name = "etcd-{version}-layer-{arch}".format(
+            arch = arch,
+            version = version,
+        ),
+        directory = "/opt/etcd-v{version}-linux-{arch}/".format(
+            arch = arch,
+            version = version,
+        ),
+        files = [
+            "//:etcd-v{version}-linux-{arch}_etcdctl".format(
+                arch = arch,
+                version = version,
+            ),
+            "//:etcd-v{version}-linux-{arch}_etcd".format(
+                arch = arch,
+                version = version,
+            ),
+        ],
+    )
+    for (arch, version) in supported_etcd_arch_and_version()
+]
+
+# We build a base layer that includes all the versions of etcd we want to ship
+# On non-amd64 we exclude the older versions of etcd
 container_image(
     name = "etcd-manager-base",
-    base = "@debian-base-amd64//image",
+    base = select({
+        "@io_bazel_rules_go//go/platform:amd64": "@debian-hyperkube-base-amd64//image",
+        "@io_bazel_rules_go//go/platform:arm64": "@debian-hyperkube-base-arm64//image",
+    }),
     directory = "/opt",
-    layers = [
-        "etcd-2-2-1-layer",
-        "etcd-3-1-12-layer",
-        "etcd-3-2-18-layer",
-        "etcd-3-2-24-layer",
-        "etcd-3-3-10-layer",
-        "etcd-3-3-13-layer",
-        "etcd-3-3-17-layer",
-        "etcd-3-4-3-layer",
-        "etcd-3-4-13-layer",
-    ],
+    layers = select({
+        "@io_bazel_rules_go//go/platform:amd64": [
+            "etcd-2.2.1-layer-amd64",
+            "etcd-3.1.12-layer-amd64",
```

Comment:
We may be able to generate (most) of this from supported_etcd_arch_and_version ... an idea for the future :-)